src/main/java/crazypants/enderio/teleport/TravelController.java

```diff
@@ -371,49 +411,37 @@ public boolean doTeleport(EntityPlayer player) {
             // we test against our feets location
             sample.y -= playerHeight;
 
-            if (doBlinkAround(player, source, sample, false)) {
-                return true;
+            Optional<BlockCoord> destinationOptional = findNearbyDestination(player, source, sample);
+            if (destinationOptional.isPresent()) {
+                return destinationOptional;
             }
         }
-        return false;
+        return Optional.empty();
     }
 
-    /** Returns empty optional if no travel destination was found. */
+    /** Finds a long-distance travel anchor, or returns empty optional if unsuccessful. */
     public Optional<BlockCoord> findTravelDestination(EntityPlayer player, TravelSource source) {
```

Comment:
I prefer to return `Optional` rather than `null` from methods that may not return a value, in order to force explicit handling of the `null` case, as well as being completely clear about the return value possibly being `null`. Isn't that the main point of `Optional`?

Yeah, I think the methods are in a rather nonsensical order, since I think they've kind of gotten added over time 😓. I've moved this method (and `findTeleportDestination()`, right above it) as you requested.

Reply:
`Optional` generally makes sense when both `null` and "unset"/"unknown" are valid return values with different semantics. Here the result is either a valid `BlockCoord`, or "no valid destination found". There is no need for a different third state.

Are these methods ever used (or meant to be used) outside of `TravelController`? Maybe they should be made private so that you don't need to worry about other code using it incorrectly.

Reply:
Fair enough, your reasons make sense here.